components/director/internal/domain/tenant/repository.go

```diff
@@ -370,6 +356,30 @@ func (r *pgRepository) GetLowestOwnerForResource(ctx context.Context, resourceTy
 	return dest.TenantID, nil
 }
 
+func (r *pgRepository) GetBySubscribedRuntimes(ctx context.Context) ([]*model.BusinessTenantMapping, error) {
+	persist, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var entityCollection tenant.EntityCollection
+
+	query := `
```

Comment:
The query should use the lister with the respective criteria. 

Also, the names of the fields and the tables should be parametrized accordingly (with the names in the file above and the ones for tenant accesses tables)

See: https://github.com/kyma-incubator/compass/blob/3e8dee4ba319971e9694d22680115e81e7059f39/components/director/pkg/resource/resource.go#L118

 and 

https://github.com/kyma-incubator/compass/blob/5a8eaeb471b31f97fb2d0e79a3286b323d76c9da/components/director/internal/repo/tenant_access_m2m.go#L18